h2o-py/tests/testdir_misc/pyunit_metric_accessors.py

```diff
@@ -420,9 +436,19 @@ def metric_accessors():
     response_col = "cylinders"
     distribution = "multinomial"
     predictors = ["displacement","power","weight","acceleration","year"]
-    gbm.distribution="multinomial"
+    gbm = H2OGradientBoostingEstimator(nfolds=3, distribution=distribution, fold_assignment="Random")
     gbm.train(x=predictors,y=response_col, training_frame=train, validation_frame=valid)
 
+    # using list from http://docs.h2o.ai/h2o/latest-stable/h2o-docs/performance-and-prediction.html#classification
+    # + common ones
+    for metric in ['logloss', 'mse', 'rmse', 'mean_per_class_error']:
+        val = getattr(gbm, metric)()
+        assert isinstance(val, float), "expected a float for metric {} but got {}".format(metric, val)
```

Comment:
please again admire how `model.mean_per_class_error()` returns a `float` for `multinomial`, but an array for `binomial`...

Reply:
This is an error, right?  You can actually get away with one value for binomial but for multinomial, it must be an array.  Please write a JIRA for this when you have a chance.

Reply:
> please again admire how model.mean_per_class_error() returns a float for multinomial, but an array for binomial...

really? how does it return a single number for multinomial? what is that number?

Reply:
@wendycwong, @michalkurka: there are various inconsistencies here:
- "single number for multinomial" is fine, as the metric is `mean_per_class_error`, not `mean_error_per_class`, a single float is what is expected here.
- although `mean_per_class_error` returns an array for `binomial` because there, the metric methods also support a `threshold` and we always return an array of [threshold, value] even if no threshold or a single threshold was provided, making the result at the same time unexpected but making sense (it also provides the threshold for which the metric is computed).
- then there is this inconsistency among binomial metrics themselves: the ones computed globally (auc, aucpr, logloss, gini...) return a single value, but the ones that accept an "optional" threshold (F1, accuracy, mcc, mean_per_class_error...) return an array or arrays as explained above.

This is not completely wrong or buggy, it is just that the API is not very user-friendly...

Reply:
I can write a ticket to improve the usability here., but I don't think this can be done without sacrificing a minimum backward compatibility.

Reply:
seb:  I wrote a JIRA for fixing the Gini: https://0xdata.atlassian.net/browse/PUBDEV-6888.

Thank you in advance!